File: csharp/Domain/Models/RallyTeam.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Models
{
    public class RallyTeam : Identifiable<long>
    {
        public String TeamName { get; set; }

        public RallyTeam(string teamName)
        {
            TeamName = teamName;
        }

        public override bool Equals(object? obj)
        {
            return obj is RallyTeam team &&
                   base.Equals(obj) &&
                   Id == team.Id &&
                   TeamName == team.TeamName;
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(base.GetHashCode(), Id, TeamName);
        }
    }
}

File: java/domain/src/main/java/models/RallyTeam.java
package models;

import java.util.Objects;

public class RallyTeam extends Identifiable<Long> {
    private String teamName;

    public RallyTeam(String teamName) {
        this.teamName = teamName;
    }

    public String getTeamName() {
        return teamName;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof RallyTeam)) return false;
        if (!super.equals(o)) return false;
        RallyTeam rallyTeam = (RallyTeam) o;
        return Objects.equals(teamName, rallyTeam.teamName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), teamName);
    }
}
